src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -147,8 +146,13 @@ private GenericRecord getPostHookEvent(HookContext hookContext, EventStatus stat
         .ifPresent(
             applicationId -> {
               recordBuilder.set("YarnApplicationId", applicationId.toString());
-              retrieveYarnQueueName(conf, applicationId)
-                  .ifPresent(queue -> recordBuilder.set("Queue", queue));
+              yarnApplicationRetriever
```

Comment:
nit: I am more inclined to `peek` and `flatMap` in these cases

Reply:
flatMap returns  an optional and here we just consume the value. I could move `YarnApplicationId` under `yarnApplicationRetriever`, i.e.:

```
ApplicationIdRetriever.determineApplicationId(conf, getExecutionMode(plan))
        .flatMap(applicationId -> yarnApplicationRetriever.retrieve(conf, applicationId))
        .ifPresent(
            applicationReport -> {
              recordBuilder.set("YarnApplicationId", applicationReport.getApplicationId().toString());
              recordBuilder.set("HiveHostName", applicationReport.getHost());
              recordBuilder.set("Queue", applicationReport.getQueue());
            });
```

But then if there an error in yarnApplicationRetriever, we'll lose `YarnApplicationId` too. I think it might be useful to debug such issues – search for queries which have application id, but no queue.